planner/core/exhaust_physical_plans.go

```diff
@@ -262,18 +262,26 @@ func (p *LogicalJoin) getHashJoins(prop *property.PhysicalProperty) []PhysicalPl
 	}
 	joins := make([]PhysicalPlan, 0, 2)
 	switch p.JoinType {
-	case SemiJoin, AntiSemiJoin, LeftOuterSemiJoin, AntiLeftOuterSemiJoin, LeftOuterJoin:
-		joins = append(joins, p.getHashJoin(prop, 1))
+	case SemiJoin, AntiSemiJoin, LeftOuterSemiJoin, AntiLeftOuterSemiJoin:
+		joins = append(joins, p.getHashJoin(prop, 1, false))
+	case LeftOuterJoin:
+		joins = append(joins, p.getHashJoin(prop, 1, false))
+		if len(p.EqualConditions) > 0 {
+			joins = append(joins, p.getHashJoin(prop, 1, true))
+		}
 	case RightOuterJoin:
-		joins = append(joins, p.getHashJoin(prop, 0))
+		joins = append(joins, p.getHashJoin(prop, 0, false))
+		if len(p.EqualConditions) > 0 {
+			joins = append(joins, p.getHashJoin(prop, 0, true))
```

Comment:
We can't directly get the `outerHashJoin`. The `innerHashJoin` can keep order for `prop.items` by send them to the outer child side and also keep outer order during the hash join. But `outerHashJoin` can't achieve it. It has no ability to "keep outer order during the hash join".

Reply:
```
func (p *LogicalJoin) getHashJoins(prop *property.PhysicalProperty) []PhysicalPlan {
	if !prop.IsEmpty() { // hash join doesn't promise any orders
		return nil
	}
```
This code lays at the beginning of this function, which means the hash join does not promose any orders.